parsl/curvezmq.py

```diff
@@ -0,0 +1,195 @@
+import os
+from abc import ABCMeta, abstractmethod
+from typing import Optional, Tuple, Union
+
+import zmq
+import zmq.auth
+from zmq.auth.thread import ThreadAuthenticator
+
+
+def create_certificates(base_dir: Union[str, os.PathLike]):
+    """Create server and client certificates in a private directory.
+
+    Parameters
+    ----------
+    base_dir : str | os.PathLike
+        Parent directory of the private certificates directory.
+    """
+    cert_dir = os.path.join(base_dir, "certificates")
+    os.makedirs(cert_dir, mode=0o700, exist_ok=True)
```

Comment:
Given the direction to create certificates anew each run, is `exist_ok=True` what you intend here?

Reply:
I don't think it's needed here, but I thought you asked me to add it during an out-of-band discussion. I must've misunderstood, so I'll remove it.

Reply:
Scratch that; we want this in place for Globus Compute, which will overwrite the cert files each time a user starts an endpoint. The same goes for any other project that organizes its run directory differently.